src/helpers.ts

```diff
@@ -0,0 +1,91 @@
+import type { ABI, ABIFunction, ABIInOut } from "./abi.js";
+
+// Regular expression to match tuple types.
+// Example: tuple, tuple[], tuple[5]
+const reTupleType = /^tuple(\[\d*\])?$/;
+
+/**
+ * Fills tuple component's empty names in an ABI with generated names
+ *
+ * @example
+ * Input: {
+ *   "type": "function",
+ *   "selector": "0x95d376d7",
+ *   "payable": false,
+ *   "stateMutability": "payable",
+ *   "inputs": [
+ *     {
+ *       "type": "tuple",
+ *       "name": "",
+ *       "components": [
+ *         { "type": "uint32", "name": "" },
+ *         { "type": "bytes", "name": "" },
+ *         { "type": "bytes32", "name": "" },
+ *         { "type": "uint64", "name": "" },
+ *         { "type": "address", "name": "" }
+ *       ]
+ *     },
+ *     { "type": "bytes", "name": "" }
+ *   ],
+ *   "sig": "assignJob((uint32,bytes,bytes32,uint64,address),bytes)",
+ *   "name": "assignJob",
+ *   "constant": false
+ * }
+ *
+ * Output: {
+ *   "type": "function",
+ *   "selector": "0x95d376d7",
+ *   "payable": false,
+ *   "stateMutability": "payable",
+ *   "inputs": [
+ *     {
+ *       "type": "tuple",
+ *       "name": "",
+ *       "components": [
+ *         { "type": "uint32", "name": "field0" },
+ *         { "type": "bytes", "name": "field1" },
+ *         { "type": "bytes32", "name": "field2" },
+ *         { "type": "uint64", "name": "field3" },
+ *         { "type": "address", "name": "field4" }
+ *       ]
+ *     },
+ *     { "type": "bytes", "name": "" }
+ *   ],
+ *   "sig": "assignJob((uint32,bytes,bytes32,uint64,address),bytes)",
+ *   "name": "assignJob",
+ *   "constant": false
+ * }
+ *
+ * @param abi The ABI to process
+ * @returns A new ABI with tuple component names filled
+ */
+export function abiFillEmptyNames(abi: ABI): ABI {
```

Comment:
Hmm should we just put this in `abi.ts` instead of making a `helpers` just for this?

Reply:
Sound good! Updated: 6a0590f2a8fbfef7e3dedee097bfcb090aeed202